packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/User.kt

```diff
@@ -4,85 +4,84 @@
-    var reports: Collection<Report> = arrayListOf()
+    @OneToMany(mappedBy = "author", targetEntity = Asset::class)
+    @Where(clause = "ASSET_TYPE = 'Report'")
+    var reports: MutableSet<Report> = mutableSetOf()
 
-    @get:ManyToMany
-    @get:JoinTable(
-            name = "map_user_to_fav_reports",
-            joinColumns = arrayOf(JoinColumn(name = "user_id", referencedColumnName = "id")),
-            inverseJoinColumns = arrayOf(JoinColumn(name = "report_id", referencedColumnName = "id"))
+    @ManyToMany
+    @JoinTable(
+        name = "map_user_to_fav_reports",
+        joinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")],
+        inverseJoinColumns = [JoinColumn(name = "report_id", referencedColumnName = "id")]
     )
-    var favoriteReports: Collection<Report> = arrayListOf()
+    var favoriteReports: MutableSet<Report> = mutableSetOf()
 
-    @get:OneToMany(mappedBy = "author", targetEntity = Asset::class)
-    @get:Where(clause = "ASSET_TYPE = 'Dashboard'")
-    var dashboards: Collection<Dashboard> = arrayListOf()
+    @OneToMany(mappedBy = "author", targetEntity = Asset::class)
+    @Where(clause = "ASSET_TYPE = 'Dashboard'")
+    var dashboards: MutableSet<Dashboard> = mutableSetOf()
 
-    @get:UpdatePermission(expression = "is author of dashboard")
-    @get:ManyToMany
-    @get:JoinTable(
-            name = "map_editor_to_dashboard_collections",
-            joinColumns = arrayOf(JoinColumn(name = "user_id", referencedColumnName = "id")),
-            inverseJoinColumns = arrayOf(JoinColumn(name = "dashboard_collection_id", referencedColumnName = "id")))
-    var editingDashboards: Collection<Dashboard> = arrayListOf()
+    @UpdatePermission(expression = IS_DASHBOARD_AUTHOR)
+    @ManyToMany
+    @JoinTable(
+        name = "map_editor_to_dashboard_collections",
+        joinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")],
+        inverseJoinColumns = [JoinColumn(name = "dashboard_collection_id", referencedColumnName = "id")]
+    )
+    var editingDashboards: MutableSet<Dashboard> = mutableSetOf()
 
-    @get:ManyToMany
-    @get:JoinTable(
-            name = "map_user_to_fav_dashboards",
-            joinColumns = arrayOf(JoinColumn(name = "user_id", referencedColumnName = "id")),
-            inverseJoinColumns = arrayOf(JoinColumn(name = "dashboard_id", referencedColumnName = "id"))
+    @ManyToMany
+    @JoinTable(
+        name = "map_user_to_fav_dashboards",
+        joinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")],
+        inverseJoinColumns = [JoinColumn(name = "dashboard_id", referencedColumnName = "id")]
     )
-    var favoriteDashboards: Collection<Dashboard> = arrayListOf()
+    var favoriteDashboards: MutableSet<Dashboard> = mutableSetOf()
 
-    @get:OneToMany(targetEntity = Role::class)
-    override var roles: Collection<Role> = arrayListOf()
+    @OneToMany
+    override var roles: MutableSet<Role> = mutableSetOf()
```

Comment:
Who can edit what roles a user belongs to?

Reply:
At the moment, nobody. We need to add a prefab `admin` role and allow that role to make user updates (along with other things)